edx_sga/static/js/src/edx_sga.js

```diff
@@ -161,13 +170,22 @@ function StaffGradedAssignmentXBlock(runtime, element) {
             $("button#download-selected-submissions").on("click", function() {
                 var student_ids = [];
                 $submissionCheckboxes.filter(":checked").each(function() {
+                    var $submissionFilename = $(this).closest("tr").find("a.submission-filename");
+                    if (!$submissionFilename.prev("i").length) {
+                        $submissionFilename.before(submissionDownloadedHtml);
+                    }
```

Comment:
This looks very similar to the thing you're doing in the code above. You can create a function to DRY.